tasks/eva_2406/gather_release_species.py

```diff
@@ -0,0 +1,340 @@
+            url = ensembl_url + '/' + scientific_name.lower().replace(' ', '_')
+            response = requests.get(url, params={'content-type': 'application/json'})
+            data = response.json()
+            assembly_accession = str(data.get('assembly_accession'))
+            cache['scientific_name_to_ensembl'][scientific_name] = assembly_accession
+        return [str(taxid), str(scientific_name), cache['scientific_name_to_ensembl'].get(scientific_name)]
+
+    return ['NA', 'NA', 'NA']
+
+
+def find_all_eva_studies(accession_counts, private_config_xml_file):
+    metadata_uri = get_pg_metadata_uri_for_eva_profile("development", private_config_xml_file)
+    with psycopg2.connect(metadata_uri, user="evadev") as pg_conn:
+        query = (
+            'SELECT DISTINCT a.vcf_reference_accession, pt.taxonomy_id, p.project_accession '
+            'FROM project p '
+            'LEFT OUTER JOIN project_analysis pa ON p.project_accession=pa.project_accession '
+            'LEFT OUTER JOIN analysis a ON pa.analysis_accession=a.analysis_accession '
+            'LEFT OUTER JOIN project_taxonomy pt ON p.project_accession=pt.project_accession '
+            'ORDER BY pt.taxonomy_id, a.vcf_reference_accession'
+        )
+        data = []
+        for assembly, tax_id, study in filter_studies(get_all_results_for_query(pg_conn, query)):
+            taxid_from_ensembl, scientific_name, ensembl_assembly_from_taxid = retrieve_current_ensembl_assemblies(tax_id)
+            _, _, ensembl_assembly_from_assembly = retrieve_current_ensembl_assemblies(assembly)
+
+            count_ssid = 0
+            if study in accession_counts:
+                assembly_from_mongo, taxid_from_mongo, project_accession, count_ssid = accession_counts.pop(study)
+                if assembly_from_mongo != assembly:
+                    logger.error(
+                        'For study %s, assembly from accessioning (%s) is different'
+                        ' from assembly from metadata (%s) database.', study, assembly_from_mongo, assembly
+                    )
+                if taxid_from_mongo != tax_id:
+                    logger.error(
+                        'For study %s, taxonomy from accessioning (%s) is different'
+                        ' from taxonomy from metadata (%s) database.', study, taxid_from_mongo, tax_id
+                    )
+            data.append({
+                'Source': 'EVA',
+                'Assembly': assembly,
+                'Taxid': tax_id,
+                'Scientific Name': scientific_name,
+                'Study': study,
+                'Number Of Variants (submitted variants)': count_ssid or 0,
+                'Ensembl assembly from taxid': ensembl_assembly_from_taxid,
+                'Ensembl assembly from assembly': ensembl_assembly_from_assembly,
+                'Target Assembly': ensembl_assembly_from_taxid or ensembl_assembly_from_assembly
```

Comment:
also, is it ever the case that the ensembl assembly from taxid vs from assembly are both present and different?

Reply:
Yes it does happen and in this case I prefer the one from the taxonomy id because it would be the official Ensembl assembly for that species as opposed to whatever the submitter used.